### PR TITLE
fix: bump Feast to 0.62 in manifests and runtime pylocks

### PR DESCRIPTION
**Pull request by jiridanek**
## Summary
- Update ODH manifest annotations for pytorch, rocm-pytorch, and tensorflow to declare Feast 0.62 (missed by c44d687e2)
- Update runtime pylock files (pytorch, tensorflow, rocm-pytorch) from feast 0.61.0 to 0.62.0 using wheels already available on the RH registry
- Regenerate RHOAI manifest annotations via `update_imagestream_annotations_from_pylock.py`

## Test plan
- [ ] `make test` passes (specifically `test_image_pyprojects` no longer fails on Feast version mismatch)

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Dependencies**
  * Updated Feast library from version 0.61 to 0.62 across PyTorch, TensorFlow, and ROCm notebook image variants for both OpenDataHub and RHOAI deployments.

### CODE REVIEW

File: manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml
Comment (jiridanek) [inline manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai explain how this got broken/out of sync?

> cause: Commit c44d687e2 ("bump feast to 0.62.0") updated the jupyter pylock files to feast 0.62.0 but only updated the datascience ODH manifest and all RHOAI manifests. It missed updating the ODH manifests for pytorch, tensorflow, and rocm-pytorch. Additionally, the runtime pylock files still pinned feast 0.61.0 from the RH internal registry (which now has 0.62.0 available).

what should've happened? did we ignore red ci when we should not? or what went wrong?

File: runtimes/rocm-pytorch/ubi9-python-3.12/uv.lock.d/pylock.rocm.toml [hunk 1]
```diff
@@ -288,9 +288,9 @@ wheels = [{ url = "https://packages.redhat.com/api/pulp-content/public-rhai/rhoa
 
 [[packages]]
 name = "feast"
-version = "0.61.0"
+version = "0.62.0"
 marker = "implementation_name == 'cpython' and sys_platform == 'linux'"
-wheels = [{ url = "https://packages.redhat.com/api/pulp-content/public-rhai/rhoai/3.4/rocm6.4-ubi9/feast-0.61.0-12-py3-none-any.whl", upload-time = 2026-03-11T00:37:21Z, size = 8205934, hashes = { sha256 = "f0c36edacd29273a0bbf28aaec7b971d823158a78ea1ee0eff37417e0a52a7e7" } }]
+wheels = [{ url = "https://packages.redhat.com/api/pulp-content/public-rhai/rhoai/3.4/rocm6.4-ubi9/feast-0.62.0-12-py3-none-any.whl", upload-time = 2026-04-09T01:28:59Z, size = 7993843, hashes = { sha256 = "71004070a9e20f77c3a3e097948b7aa5896a178693fa1b30f7184b6f30d6cda5" } }]
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply runtimes/rocm-pytorch/ubi9-python-3.12/uv.lock.d/pylock.rocm.toml:293]: I don't know why; you tell me? Anyways, this makes the `gmake test` pass!

File: manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml
Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai explain how this got broken/out of sync?

> cause: Commit c44d687 ("bump feast to 0.62.0") updated the jupyter pylock files to feast 0.62.0 but only updated the datascience ODH manifest and all RHOAI manifests. It missed updating the ODH manifests for pytorch, tensorflow, and rocm-pytorch. Additionally, the runtime pylock files still pinned feast 0.61.0 from the RH internal registry (which now has 0.62.0 available).

what should've happened? did we ignore red ci when we should not? or what went wrong?

File: runtimes/rocm-pytorch/ubi9-python-3.12/uv.lock.d/pylock.rocm.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiridanek) [reply runtimes/rocm-pytorch/ubi9-python-3.12/uv.lock.d/pylock.rocm.toml:293]: @coderabbitai https://packages.redhat.com/api/pulp-content/public-rhai/rhoai/3.4/cuda13.0-ubi9/feast-0.62.0-8-py3-none-any.whl#sha256=163844d12588ee102a7fb12a514b7498f2a2621ac8cd35c502c0feaffa1f2d6b is live, I see it at https://console.redhat.com/api/pypi/public-rhai/rhoai/3.4/cuda13.0-ubi9/simple/feast/

File: manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml
Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai if
* https://github.com/opendatahub-io/notebooks/pull/3368

is culprit, then the checks for it https://github.com/opendatahub-io/notebooks/actions/runs/24356906035/job/71126106482 all passed! find out true reason / cause

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: > validation-of-sw-versions-in-imagestreams — failed

@coderabbitai that's not checking the PR, that is checking previous tags of image, not current one

> test_image_manifests_version_alignment (inside code-static-analysis) was masked by the exception tuple change in the same commit — ("Feast", ("0.60", "0.61")) → ("Feast", ("0.61", "0.62")) — so that specific cross-imagestream alignment check passed silently even with ODH manifests still at 0.61.

aha, that is useful to know; so, when this was changed to cause test failure? that PR is culprit!

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai But now the same test is failing on main, so, something must have unmasked this.

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: > The structural fix would be: don't widen the exception tuple in the same PR as the bump. The tuple should only be widened after the old version is fully retired, or alternatively, a separate stricter check could assert that no current (N) imagestream tag is still on the old version once a bump lands.

The tuple is only used when checking N tags, btw; the `gmake test` only looks at current images

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai but the lockfile update
* https://github.com/opendatahub-io/notebooks/pull/3393

the code-static-analysis checks were also still passing, https://github.com/opendatahub-io/notebooks/actions/runs/24510595116/job/71640382496

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai pytest-tests job is part of the code-static-analysis workflow

> It's worth checking what pytest-tests reported for PR #3393 specifically, and whether test_image_pyprojects scopes its checks to runtime pylocks or only jupyter image pylocks.

please, do

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: > Something between PR #3393 merging and now caused the test to stop hiding the failure — likely the manifest_name_to_pip refactoring that was done in an intermediate PR, which may have fixed a subtle lookup failure (for a different package) that was previously masking the subtest result propagation.

@coderabbitai please bisect it

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: > The smoking gun here would be the actual pytest-tests output for PR #3393's run — the job passed at the top level, but the subtest details (which aren't surfaced in the GitHub Actions pass/fail badge) may have shown the failures.

@coderabbitai here are the https://github.com/opendatahub-io/notebooks/pull/3393 pytest-tests logs, please resolve doubts

* [job-logs.txt](https://github.com/user-attachments/files/26840806/job-logs.txt)
* [logs_64898163723.zip](https://github.com/user-attachments/files/26840818/logs_64898163723.zip) (the same, zipped)

Comment (jiridanek) [reply manifests/odh/base/jupyter-pytorch-notebook-imagestream.yaml:1]: @coderabbitai so the problem is that `gmake refresh-lock-files` does not update also manifests, and when lock files are updated without updating manifests, and the PR check is ignored, we get failing main?

**Comment by atheo89** [comment]
feast version is updated: https://github.com/search?q=repo%3Aopendatahub-io%2Fnotebooks+Feast&type=code&p=3
